src/lib/dapp/DAppSigner.ts

```diff
@@ -147,25 +163,31 @@ export class DAppSigner implements Signer {
     data: Uint8Array[],
     signOptions?: Record<string, any>,
   ): Promise<SignerSignature[]> {
-    const { signatureMap } = await this.request<SignTransactionResult['result']>({
-      method: HederaJsonRpcMethod.SignMessage,
-      params: {
-        signerAccountId: this._signerAccountId,
-        message: Uint8ArrayToBase64String(data[0]),
-      },
-    })
-
-    const sigmap = base64StringToSignatureMap(signatureMap)
+    this.logger.debug('Signing data')
+    try {
+      const { signatureMap } = await this.request<SignTransactionResult['result']>({
+        method: HederaJsonRpcMethod.SignMessage,
+        params: {
+          signerAccountId: this._signerAccountId,
+          message: Uint8ArrayToBase64String(data[0]),
```

Comment:
Why is this encoded?  HIP-20 spec expects a string message - not anything encoded.  HashPack et. al. ought to be able to display this string message to the user as-is.

Reply:
I can take a look at this, but just to clarify, this code hasn't changed from previous releases. We've just added a try catch here for graceful error handling

Reply:
@bugbytesinc I've added a change here that will accept either base64 or utf-8 including on the data being signed, utf-8 is the default and ensures HashPack shows a plaintext string. 

Note, we cannot modify the implementation of `sign` to accept a string because that would break the `Signer` Interface.